File: PageTitle/Designer/PageTitleDesigner.js
Type.registerNamespace("RandomSiteControls.PageTitle.Designer");

RandomSiteControls.PageTitle.Designer.PageTitleDesigner = function (element) {
    /* Initialize PageTitleText fields */
    this._pageTitleText = null;
    
    /* Calls the base constructor */
    RandomSiteControls.PageTitle.Designer.PageTitleDesigner.initializeBase(this, [element]);
}

RandomSiteControls.PageTitle.Designer.PageTitleDesigner.prototype = {
    /* --------------------------------- set up and tear down --------------------------------- */
    initialize: function () {
        /* Here you can attach to events or do other initialization */
        RandomSiteControls.PageTitle.Designer.PageTitleDesigner.callBaseMethod(this, 'initialize');
    },
    dispose: function () {
        /* this is the place to unbind/dispose the event handlers created in the initialize method */
        RandomSiteControls.PageTitle.Designer.PageTitleDesigner.callBaseMethod(this, 'dispose');
    },

    /* --------------------------------- public methods ---------------------------------- */

    findElement: function (id) {
        var result = jQuery(this.get_element()).find("#" + id).get(0);
        return result;
    },

    /* Called when the designer window gets opened and here is place to "bind" your designer to the control properties */
    refreshUI: function () {
        var controlData = this._propertyEditor.get_control(); /* JavaScript clone of your control - all the control properties will be properties of the controlData too */

        /* RefreshUI PageTitleText */
        jQuery(this.get_pageTitleText()).val(controlData.PageTitleText);
    },

    /* Called when the "Save" button is clicked. Here you can transfer the settings from the designer to the control */
    applyChanges: function () {
        var controlData = this._propertyEditor.get_control();

        /* ApplyChanges PageTitleText */
        controlData.PageTitleText = jQuery(this.get_pageTitleText()).val();
    },

    /* --------------------------------- event handlers ---------------------------------- */

    /* --------------------------------- private methods --------------------------------- */

    /* --------------------------------- properties -------------------------------------- */

    /* PageTitleText properties */
    get_pageTitleText: function () { return this._pageTitleText; }, 
    set_pageTitleText: function (value) { this._pageTitleText = value; }
}

RandomSiteControls.PageTitle.Designer.PageTitleDesigner.registerClass('RandomSiteControls.PageTitle.Designer.PageTitleDesigner', Telerik.Sitefinity.Web.UI.ControlDesign.ControlDesignerBase);

File: PageTitle/Designer/PageTitleDesigner.cs
using System;
using System.Linq;
using System.Web.UI;
using Telerik.Sitefinity.Web.UI;
using Telerik.Sitefinity.Web.UI.ControlDesign;
using System.Collections.Generic;

[assembly: WebResource(RandomSiteControls.PageTitle.Designer.PageTitleDesigner.scriptReference, "application/x-javascript")]
namespace RandomSiteControls.PageTitle.Designer
{
    /// <summary>
    /// Represents a designer for the <typeparamref name="RandomSiteControls.PageTitle.PageTitle"/> widget
    /// </summary>
    public class PageTitleDesigner : ControlDesignerBase
    {
        #region Properties
        /// <summary>
        /// Obsolete. Use LayoutTemplatePath instead.
        /// </summary>
        protected override string LayoutTemplateName
        {
            get
            {
                return string.Empty;
            }
        }

        /// <summary>
        /// Gets the layout template's relative or virtual path.
        /// </summary>
        public override string LayoutTemplatePath
        {
            get
            {
                if (string.IsNullOrEmpty(base.LayoutTemplatePath))
                    return PageTitleDesigner.layoutTemplatePath;
                return base.LayoutTemplatePath;
            }
            set
            {
                base.LayoutTemplatePath = value;
            }
        }

        protected override HtmlTextWriterTag TagKey
        {
            get
            {
                return HtmlTextWriterTag.Div;
            }
        }
        #endregion

        #region Control references
        /// <summary>
        /// Gets the control that is bound to the PageTitleText property
        /// </summary>
        protected virtual Control PageTitleText
        {
            get
            {
                return this.Container.GetControl<Control>("PageTitleText", true);
            }
        }

        #endregion

        #region Methods
        protected override void InitializeControls(Telerik.Sitefinity.Web.UI.GenericContainer container)
        {
            // Place your initialization logic here
        }
        #endregion

        #region IScriptControl implementation
        /// <summary>
        /// Gets a collection of script descriptors that represent ECMAScript (JavaScript) client components.
        /// </summary>
        public override System.Collections.Generic.IEnumerable<System.Web.UI.ScriptDescriptor> GetScriptDescriptors()
        {
            var scriptDescriptors = new List<ScriptDescriptor>(base.GetScriptDescriptors());
            var descriptor = (ScriptControlDescriptor)scriptDescriptors.Last();

            descriptor.AddElementProperty("pageTitleText", this.PageTitleText.ClientID);

            return scriptDescriptors;
        }

        /// <summary>
        /// Gets a collection of ScriptReference objects that define script resources that the control requires.
        /// </summary>
        public override System.Collections.Generic.IEnumerable<System.Web.UI.ScriptReference> GetScriptReferences()
        {
            var scripts = new List<ScriptReference>(base.GetScriptReferences());
            scripts.Add(new ScriptReference(PageTitleDesigner.scriptReference, typeof(PageTitleDesigner).Assembly.FullName));
            return scripts;
        }
        #endregion

        #region Private members & constants
        public static readonly string layoutTemplatePath = "~/SitefinitySteve/RandomSiteControls.PageTitle.Designer.PageTitleDesigner.ascx";
        public const string scriptReference = "RandomSiteControls.PageTitle.Designer.PageTitleDesigner.js";
        #endregion
    }
}
 
